charmcraft/const.py

```diff
@@ -98,6 +98,20 @@ def __str__(self) -> str:
         return str(self.value)
 
 
+GO_ARCH_TO_CHARM_ARCH = {
```

Comment:
what about the "variant"? https://github.com/canonical/rockcraft/blob/main/rockcraft/oci.py#L102

Reply:
We don't worry about variants, as charmhub doesn't. If the OCI contains any image that's compatible with the arch, we tell charmhub that we support that arch.